core/server/common/src/main/webapp/browse-body.jsp

```diff
@@ -17,35 +17,17 @@
 
 <jsp:include page="header-scripts.jsp" />
 
-<script>
-  var currentDir = "<%= request.getAttribute("currentPath").toString() %>";
-  function getInputPath() {
-    var path = $.trim($("#pathInput").val());
-    return path;
-  }
-
-  function changeDir() {
-    var path = getInputPath();
-    path = encodeURI(path);
-    window.location.href = "./browse?path=" + path;
-  }
+<link rel="stylesheet" href="css/ui-grid.css" type="text/css">
+<script src="js/angular.js"></script>
+<script src="js/ui-grid.js"></script>
+<script src="js/browse.js" ></script>
 
-  $(document).ready(function () {
-    if (base !== "./browse") {
-      $("#pathNav").hide();
-    } else {
-      // set currentDir as default value for #pathInput.
-      $("#pathInput").val(currentDir);
-      // when clicking #goBtn or enter the return key, change directory.
-      $("#goBtn").click(changeDir);
-      $("#pathInput").keydown(function (e) {
-        if (e.keyCode === 13) {
-          changeDir();
-        }
-      });
-    }
-  });
+<script>
+    var currentDir = "<%= request.getAttribute("currentPath").toString()%>";
+    var base = "<%= (request.getAttribute("baseUrl") == null) ? "./browse" : request.getAttribute("baseUrl").toString() %>";
+    var showPermissions = "<%= request.getAttribute("showPermissions")%>";
```

Comment:
Almost done except `base`, this value must be got from request, and in javascript file cannot get attribute from request as far as I know. If there is another way, please tell me, Thank you.